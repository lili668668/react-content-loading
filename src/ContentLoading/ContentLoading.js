import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Color from 'color'
import LinearGradient from '../utils/linearGradient'

const StyledContainer = styled.div`
  overflow: hidden;
  height: ${props => props.height};
  width: ${props => props.width};
  background: ${props => props.placeholderColor};
`

const StyledLoading = styled.div`
  height: ${props => props.height};
  width: ${props => props.width};
  background: ${props => props.loadingBackground};

  -moz-animation: 2s ease-in-out 0.5s infinite loading;
  animation: 2s ease-in-out 0.5s infinite loading;

  @-moz-keyframes loading {
    from {
      margin-left: -100%;
    }
    to {
      margin-left: 100%;
    }
  }

  @keyframes loading {
    from {
      margin-left: -100%;
    }
    to {
      margin-left: 100%;
    }
  }
`

class ContentLoading extends Component {
  render() {
    const { placeholderColor, loadingColor, height, width } = this.props

    const placeholderColorInstance = Color(placeholderColor)
    const loadingColorInstance = Color(loadingColor)
    const loadingBackground = new LinearGradient()
      .to('right')
      .appendColorStop(placeholderColorInstance.rgb().alpha(0.1).toString(), '5%')
      .appendColorStop(loadingColorInstance.rgb().alpha(0.8).toString(), '15%')
      .appendColorStop(loadingColorInstance.rgb().alpha(0.9).toString(), '50%')
      .appendColorStop(loadingColorInstance.rgb().alpha(0.8).toString(), '85%')
      .appendColorStop(placeholderColorInstance.rgb().alpha(0.1).toString(), '95%')
      .toString()

    return (
      <StyledContainer height={height} width={width} placeholderColor={placeholderColor}>
        <StyledLoading height={height} width={width} loadingBackground={loadingBackground} />
      </StyledContainer>
    )
  }
}

ContentLoading.propTypes = {
  /** The color of loading background */
  placeholderColor: PropTypes.string,
  /** The color of loading frontground */
  loadingColor: PropTypes.string,
  /** The height of loading */
  height: PropTypes.string,
  /** The width of loading */
  width: PropTypes.string
}

ContentLoading.defaultProps = {
  placeholderColor: '#efefef',
  loadingColor: '#e6e6e6',
  height: '300px',
  width: '300px'
}

export default ContentLoading
