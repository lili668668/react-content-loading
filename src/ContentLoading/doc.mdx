---
name: ContentLoading
menu: Components
route: /components/content-loading
---

import { Playground, PropsTable } from 'docz'
import ContentLoading from './ContentLoading'

# ContentLoading

## Description

<!-- TODO -->

## Props
<PropsTable of={ContentLoading} />

## Basic usage

`import ContentLoading from 'react-content-loading/lib/ContentLoading'` or `import { ContentLoading } from 'react-content-loading'`

<Playground>
  <ContentLoading />
</Playground>
