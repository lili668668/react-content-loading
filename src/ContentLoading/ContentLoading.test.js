import React from 'react'
import { shallow } from 'enzyme'

import ContentLoading, { getHeight, getWidth, getPlaceholderColor, getLoadingBackground } from './ContentLoading'

describe('ContentLoading', () => {
  it ('should render correctly', () => {
    const component = shallow(<ContentLoading />)
    expect(component).toMatchSnapshot()
  })
  it ('should get item of style props correctly', () => {
    const props = {
      height: '300px',
      width: '300px',
      placeholderColor: '#efefef',
      loadingBackground: 'linear-gradient(to right, rgba(239, 239, 239, 0.1) 5%, rgba(230, 230, 230, 0.8) 15%, rgba(230, 230, 230, 0.9) 50%, rgba(230, 230, 230, 0.8) 85%, rgba(239, 239, 239, 0.1) 95%)'
    }
    expect(getHeight(props)).toBe(props.height)
    expect(getWidth(props)).toBe(props.width)
    expect(getPlaceholderColor(props)).toBe(props.placeholderColor)
    expect(getLoadingBackground(props)).toBe(props.loadingBackground)
  })
})


