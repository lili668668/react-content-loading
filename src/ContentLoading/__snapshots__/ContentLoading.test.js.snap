// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentLoading should render correctly 1`] = `
<styled.div
  height="300px"
  placeholderColor="#efefef"
  width="300px"
>
  <styled.div
    height="300px"
    loadingBackground="linear-gradient(to right, rgba(239, 239, 239, 0.1) 5%, rgba(230, 230, 230, 0.8) 15%, rgba(230, 230, 230, 0.9) 50%, rgba(230, 230, 230, 0.8) 85%, rgba(239, 239, 239, 0.1) 95%)"
    width="300px"
  />
</styled.div>
`;
